File: APM/Java/Interpreter/src/ro/sandorrobertk94/domain/AssignmentStatement.java
package ro.sandorrobertk94.domain;

/**
 * Created by Robert on 10/11/2015.
 */
public class AssignmentStatement implements IStatement {
    private String variableID;
    private Expression exp;

    public AssignmentStatement() {
        this.variableID = null;
        this.exp = null;
    }

    public AssignmentStatement(String variableID, Expression exp) {
        this.variableID = variableID;
        this.exp = exp;
    }

    public String getVariableID() {
        return variableID;
    }

    public void setVariableID(String variableID) {
        this.variableID = variableID;
    }

    public Expression getExp() {
        return exp;
    }

    public void setExp(Expression exp) {
        this.exp = exp;
    }

    @Override
    public String toString() {
        return variableID + " = " + exp.toString();
    }
}

File: APM/CSharp/Interpreter/Interpreter/domain/AssignmentStatement.cs
namespace Interpreter.domain
{
    public class AssignmentStatement : IStatement
    {
        private string _variableId;
        private Expression _exp;

        public AssignmentStatement()
        {
            _variableId = null;
            _exp = null;
        }

        public AssignmentStatement(string variableId, Expression exp)
        {
            _variableId = variableId;
            _exp = exp;
        }

        public string VariableId
        {
            get { return _variableId; }
            set { _variableId = value; }
        }

        public Expression Exp
        {
            get { return _exp; }
            set { _exp = value; }
        }

        public override string ToString()
        {
            return _variableId + " = " + _exp.ToString();
        }
    }
}